services/src/main/java/org/keycloak/services/managers/AuthenticationSessionManager.java

```diff
@@ -126,6 +132,17 @@ public void setAuthSessionCookie(String authSessionId) {
         log.debugf("Set AUTH_SESSION_ID cookie with value %s", encodedAuthSessionId);
     }
 
+    /**
+     * @param authSessionId decoded authSessionId (without route info attached)
+     */
+    public void setAuthSessionIdHashCookie(String authSessionId) {
+        String authSessionIdHash = Base64.getEncoder().withoutPadding().encodeToString(HashUtils.hash(JavaAlgorithm.SHA256, authSessionId.getBytes(StandardCharsets.UTF_8)));
```

Comment:
SHA encoding is deterministic like this no? If we're trying to hide details from the cookie then this can still be easily decoded on the client to get the actual value.

Reply:
Yes, it is deterministic in the sense that the same input always produces the same output, but from the hash value, I don't see a way to trace back to the original value on client side since the `AUTH_SESSION_ID` is not available.

Reply:
Yeah I was just wondering if we might want to salt this hash as well, but perhaps it is overkill.